### PR TITLE
Python docs update: concepts section

### PR DESCRIPTION
**Pull request by abey79**
### What

Updated the pages from the Concepts section.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] ~~I've included a screenshot or gif (if applicable)~~
* [x] ~~I have tested [demo.rerun.io](https://demo.rerun.io/pr/3627) (if applicable)~~

- [PR Build Summary](https://build.rerun.io/pr/3627)
- [Docs preview](https://rerun.io/preview/2a214104cfab180cfb3128dba1e4320833ea5da5/docs) 
- [Examples preview](https://rerun.io/preview/2a214104cfab180cfb3128dba1e4320833ea5da5/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: docs/content/concepts/annotation-context.md [hunk 1]
```diff
@@ -49,11 +49,11 @@ Keypoints are currently only applicable to 2D and 3D points.
...
 
 The Annotation Context is defined as a list of Class Descriptions that define how classes are styled
-(as well as optional keypoint style & connection).
+(as well as optional keypoint style and connection).
 
 Annotation contexts are logged with:
 
-* Python: [`log_annotation_context`](https://ref.rerun.io/docs/python/latest/common/annotations/#rerun.log_annotation_context)
+* Python: [`rr.AnnotationContext`](https://ref.rerun.io/docs/python/HEAD/common/annotations/#rerun.AnnotationContext)
 * Rust: [`AnnotationContext`](https://docs.rs/rerun/latest/rerun/components/struct.AnnotationContext.html)
```
Comment (abey79) [inline docs/content/concepts/annotation-context.md:57]: Is this correct?

File: docs/content/concepts/annotation-context.md [hunk 2]
```diff
@@ -71,13 +71,11 @@ Segmentation images are single channel integer images/tensors where each pixel r
 By default, Rerun will automatically assign colors to each class id, but by defining an Annotation Context,
 you can explicitly determine the color of each class.
 
-* Python: [`log_segmentation_image`](https://ref.rerun.io/docs/python/latest/common/images/#rerun.log_segmentation_image)
+* Python: [`rr.SegmentationImage`](https://ref.rerun.io/docs/python/HEAD/common/images/#rerun.SegmentationImag)
 * Rust: Log a [`Tensor`](https://docs.rs/rerun/latest/rerun/components/struct.Tensor.html) with [`TensorDataMeaning::ClassId`](https://docs.rs/rerun/latest/rerun/components/enum.TensorDataMeaning.html#variant.ClassId)
```
Comment (abey79) [inline docs/content/concepts/annotation-context.md:75]: Is this correct?

File: docs/content/concepts/entity-component.md [hunk 3]
```diff
@@ -4,59 +4,64 @@ order: 1
...
+make sense of.
 
-code-example: extension-components
+Your entity could have any number of additional components as well. This isn't a problem. Any components that
+aren't relevant to the scene that the space view is drawing are safely ignored. Also, Rerun even allows you to log your
+own set of components, bypassing archetypes altogether.
+
+In Python this is done via [log_extension_components](https://ref.rerun.io/docs/python/latest/common/extension_components/#rerun.log_extension_components), whereas in Rust you implement the [`Component`](https://docs.rs/rerun/latest/rerun/experimental/trait.Component.html) trait.
 
+TODO: FIXME!!!
```
Comment (abey79) [inline docs/content/concepts/entity-component.md:53]: TODO: sort out the extension component / entirely custom set code

File: docs/content/concepts/annotation-context.md (hunk 1 above)
Comment (Wumpf) [reply docs/content/concepts/annotation-context.md:57]: if we write `rr.AnnotationContext` we should probably also write `rerun::AnnotationContext` for Rust 🤔

Comment (Wumpf) [reply docs/content/concepts/annotation-context.md:57]: oh and the link will drop the `components` part soon. You can link to the alpha to get a working link

Comment (Wumpf) [reply docs/content/concepts/annotation-context.md:57]: ah shoot we don't have it on the rerun namespace on the alpha yet https://docs.rs/rerun/0.9.0-alpha.6/rerun/archetypes/struct.AnnotationContext.html#

File: docs/content/concepts/annotation-context.md (hunk 2 above)
Comment (Wumpf) [reply docs/content/concepts/annotation-context.md:75]: as above https://docs.rs/rerun/0.9.0-alpha.6/rerun/archetypes/struct.SegmentationImage.html

File: docs/content/concepts/entity-component.md [hunk 4]
```diff
@@ -4,59 +4,70 @@ order: 1
...
+of one or more components associated with that entity.
 
 ## ECS Systems
-In most ECS architectures, there is a third concept we haven't touched on: Systems are processes which operate on the
-Entities based on the Components they possess.
 
-Rerun does not currently have formalized Systems, although the patterns employed by the Rerun Space Views are very much
-"System like" in their operation. Proper Systems may be a feature investigated in the future
+In most ECS architectures, there is a third concept we haven't touched on: *systems* are processes which operate on the
+entities based on the components they possess. Rerun does not currently have formalized systems, although the patterns employed by the space views are very much "System like" in their operation. Proper Systems may be a feature investigated in the future
```
Comment (Wumpf) [inline docs/content/concepts/entity-component.md:72]: I'm inclined to close 1155 and instead write something else here. How about:

```
There is a third concept we haven't touched on: *systems* are processes which operate on the
entities based on the components they possess.
Rerun is still settling on the exact form of formalized systems and outside of Rust viewer code it is not yet possible to write your own systems. However, Space Views work under the hood using a variety of systems. For more information see TODO LINK TO CUSTOMIZATION
```

File: docs/content/concepts/entity-path.md [hunk 5]
```diff
@@ -4,26 +4,28 @@ order: 1
...
 
-Rerun treats these paths as being arranged in a hierarchy with the "/" character acting as a separator between path
-elements. The conventional path semantics including concepts of "root" and "parent" / "child" generally apply.
+Note that although it is technically possible to log multiple types of archetypes on the same entity path, it's almost certainly not what you want. For example, logging a [`Points3D`](../reference/data_types/archetypes/points3d.md) point cloud on an entity path where a [`Mesh3D`](../reference/data_types/archetypes/mesh3d.md) was previously logged would overwrite the mesh's [`Position3D`](../reference/data_types/components/position3d.md) component with the point cloud's, but would leave the [`MeshProperties`](../reference/data_types/components/mesh_properties.md) component untouched. The Rerun viewer would likely be unable to display the result. See the [Entity Component](entity-component.md) section for more information.
```
Comment (Wumpf) [inline docs/content/concepts/entity-path.md:11]: `it's in many cases not what you want`

The problem here is that we force you to do that for Transforms3D and a few others

**Review by Wumpf — COMMENTED** [review]
very quick pass

File: docs/content/concepts/entity-path.md [hunk 6]
```diff
@@ -4,26 +4,28 @@ order: 1
 ---
 
 ## Entity Paths
-As mentioned in the [Entity Component](entity-component.md) overview, all Entities within Rerun have a unique _Entity Path_.
+As mentioned in the [Entity Component](entity-component.md) overview, all entities within Rerun have a unique _entity path_.
 
-The first argument to each log function is this path. Each time you log to a specific entity path you will update the entity, i.e. log a new instance of it along the timeline. Each logging to a path must be of the same type (you cannot log an image to the same path as a point cloud).
+The first argument to the [`rr.log()`](https://ref.rerun.io/docs/python/HEAD/common/logging/#rerun.log) function is this path. Each time you log to a specific entity path you will update the entity, i.e. log a new instance of it along the timeline.
```
[the remark this answers is not in the document]
Comment (abey79) [reply docs/content/concepts/entity-path.md:9]: I made that section a tad more language agnostic

File: docs/content/concepts/entity-path.md (hunk 5 above)
Comment (abey79) [reply docs/content/concepts/entity-path.md:11]: Good call. I've clarified that.

**Pull request merged**